docs/secure-cloud-function-on-foundation-3-0-0.md

```diff
@@ -0,0 +1,1415 @@
+
+To apply these Organization Policies in Parent Level (Organization or Folder level), add the code below in the `1-org` step.
+
+1. Add org policies related to Cloud Function at `1-org/envs/shared/org_policy.tf`
+
+#### Boolean policy
+
+Add policy `"cloudfunctions.requireVPCConnector"` to the [boolean_type_organization_policies](https://github.com/terraform-google-modules/terraform-example-foundation/blob/v3.0.0/1-org/envs/shared/org_policy.tf#LL27C1-L43C5) list:
+
+```hcl
+  boolean_type_organization_policies = toset([
+    "compute.disableNestedVirtualization",
+    "compute.disableSerialPortAccess",
+    "compute.disableGuestAttributesAccess",
+    "compute.skipDefaultNetworkCreation",
+    "compute.restrictXpnProjectLienRemoval",
+    "compute.disableVpcExternalIpv6",
+    "compute.setNewProjectDefaultToZonalDNSOnly",
+    "compute.requireOsLogin",
+    "sql.restrictPublicIp",
+    "sql.restrictAuthorizedNetworks",
+    "iam.disableServiceAccountKeyCreation",
+    "iam.automaticIamGrantsForDefaultServiceAccounts",
+    "iam.disableServiceAccountKeyUpload",
+    "storage.uniformBucketLevelAccess",
+    "storage.publicAccessPrevention",
+    "cloudfunctions.requireVPCConnector"
+  ])
+```
+
+#### List policies
+
+Copy the following code to file [org_policy.tf](https://github.com/terraform-google-modules/terraform-example-foundation/blob/v3.0.0/1-org/envs/shared/org_policy.tf)
+
+   ```hcl
+   /******************************************
+   Cloud Function
+   *******************************************/
+
+   module "cloudfunction_allowed_ingress" {
+     source  = "terraform-google-modules/org-policy/google"
+     version = "~> 5.1"
+
+     organization_id   = local.organization_id
+     folder_id         = local.folder_id
+     policy_for        = local.policy_for
+     policy_type       = "list"
+     allow             = ["ALLOW_INTERNAL_ONLY"]
+     allow_list_length = 1
+     constraint        = "constraints/cloudfunctions.allowedIngressSettings"
+   }
+
+   module "cloudfunction_vpc_connector_egress_settings" {
+     source  = "terraform-google-modules/org-policy/google"
+     version = "~> 5.1"
+
+     organization_id   = local.organization_id
+     folder_id         = local.folder_id
+     policy_for        = local.policy_for
+     policy_type       = "list"
+     allow             = ["ALL_TRAFFIC"]
+     allow_list_length = 1
+     constraint        = "constraints/cloudfunctions.allowedVpcConnectorEgressSettings"
+   }
+
+   module "cloudrun_allowed_ingress" {
+     source  = "terraform-google-modules/org-policy/google"
+     version = "~> 5.1"
+
+     organization_id   = local.organization_id
+     folder_id         = local.folder_id
+     policy_for        = local.policy_for
+     policy_type       = "list"
+     allow             = ["is:internal-and-cloud-load-balancing"]
+     allow_list_length = 1
+     constraint        = "constraints/run.allowedIngress"
+   }
+
+   module "cloudrun_allowed_vpc_egress" {
```

Comment:
Maybe, these org policies should be applied only at the project level, need to check

Reply:
policies will be applied at organization level and a note regarding deploying organization policies at folder and project level was added.